src/cbmc/cbmc_solvers.cpp

```diff
@@ -133,23 +146,16 @@ std::unique_ptr<cbmc_solverst::solvert> cbmc_solverst::get_dimacs()
 
 std::unique_ptr<cbmc_solverst::solvert> cbmc_solverst::get_bv_refinement()
 {
-  std::unique_ptr<propt> prop=[this]() -> std::unique_ptr<propt>
-  {
-    // We offer the option to disable the SAT preprocessor
-    if(options.get_bool_option("sat-preprocessor"))
-    {
-      no_beautification();
-      return util_make_unique<satcheckt>();
-    }
-    return util_make_unique<satcheck_no_simplifiert>();
-  }();
-
-  prop->set_message_handler(get_message_handler());
+  const bool use_simplifier=options.get_bool_option("sat-preprocessor");
```

Comment:
Suggest commenting that this definition intentionally differs from the one on line 115